File: algorithms/code/hackerrank/compare_the_triplets/compare_the_triplets.py
# https://www.hackerrank.com/challenges/compare-the-triplets

import unittest


def compare_triplets(a: list[int], b: list[int]) -> list[int]:
    result = [0, 0]
    for i in range(3):
        result[0] += 1 if a[i] > b[i] else 0
        result[1] += 1 if a[i] < b[i] else 0
    return result


class TestCode(unittest.TestCase):
    def test_example_0(self) -> None:
        self.assertEqual([1, 1], compare_triplets([5, 6, 7], [3, 6, 10]))

    def test_example_1(self) -> None:
        self.assertEqual([2, 1], compare_triplets([17, 28, 30], [99, 16, 8]))

File: algorithms/code/hackerrank/compare_the_triplets/compare_the_triplets.go
// Package comparethetriplets implements https://www.hackerrank.com/challenges/compare-the-triplets
package comparethetriplets

// CompareTriplets - implements the solution to the problem
func CompareTriplets(a []int32, b []int32) []int32 {
	result := []int32{0, 0}
	for i := range 3 {
		if a[i] > b[i] {
			result[0]++
		}
		if a[i] < b[i] {
			result[1]++
		}
	}
	return result
}
